Tests/Realm.Tests/Database/CollectionTests.cs

```diff
@@ -411,6 +414,258 @@ public void Results_GetFiltered_SanityTest()
+                }
+            }
+
+            private static object? TryToConvert(RealmValue value, Converter converter)
+            {
+                try
+                {
+                    return converter(value);
+                }
+                catch
+                {
+                    return null;
+                }
+            }
+
+            private static object? TryToConvert(object value, Type targetType)
+            {
+                try
+                {
+                    return Convert.ChangeType(value, targetType);
+                }
+                catch
+                {
+                    return null;
+                }
+            }
+        }
+
+        public static IEnumerable<StringQueryTestData> StringQueryTestCases_GeneralCases()
+        {
+            yield return new StringQueryTestData(nameof(AllTypesObject.CharProperty), 'c', 'b');
+            yield return new StringQueryTestData(nameof(AllTypesObject.ByteProperty), 0x5, 0x4);
+            yield return new StringQueryTestData(nameof(AllTypesObject.Int16Property), 5, 4);
+            yield return new StringQueryTestData(nameof(AllTypesObject.Int32Property), 34, 42);
+            yield return new StringQueryTestData(nameof(AllTypesObject.Int64Property), 74L, 23L);
+            yield return new StringQueryTestData(nameof(AllTypesObject.SingleProperty), 3.0f, 2.0f);
+            yield return new StringQueryTestData(nameof(AllTypesObject.DoubleProperty), 4.0, 2.0);
+            yield return new StringQueryTestData(nameof(AllTypesObject.BooleanProperty), true, false);
+            yield return new StringQueryTestData(nameof(AllTypesObject.DateTimeOffsetProperty), new DateTimeOffset(1956, 6, 1, 0, 0, 0, TimeSpan.Zero), new DateTimeOffset(2020, 8, 2, 0, 0, 0, TimeSpan.Zero));
+            yield return new StringQueryTestData(nameof(AllTypesObject.DecimalProperty), 0.9999999999999999999999999999, 0.3333333333333333333333333333);
+            yield return new StringQueryTestData(nameof(AllTypesObject.Decimal128Property), new Decimal128(564.42343424323), new Decimal128(666.42300000003));
+            yield return new StringQueryTestData(nameof(AllTypesObject.ObjectIdProperty), new ObjectId("5f64cd9f1691c361b2451d96"), new ObjectId("5ffffffffffff22222222222"));
+            yield return new StringQueryTestData(nameof(AllTypesObject.GuidProperty), new Guid("0f8fad5b-d9cb-469f-a165-70867728950e"), new Guid("0ffffffb-dddd-4444-aaaa-70000000000e"));
+            yield return new StringQueryTestData(nameof(AllTypesObject.StringProperty), "hello world", "no salute");
+            yield return new StringQueryTestData(nameof(AllTypesObject.ByteArrayProperty), new byte[] { 0x5, 0x4, 0x3, 0x2, 0x1 }, new byte[] { 0x1, 0x1, 0x1 });
+        }
+
+        public static IEnumerable<StringQueryTestData> StringQueryTestCases_NumbericValues_ImplicitConvertion_Matches()
```

Comment:
Typo: `Numberic` -> `Numeric`

Reply:
Also, instead of reusing `StringQueryTestData`, perhaps create a different type for these - it will allow you to be more explicit about what the test is doing:
1. Instead of overloading the meaning of `NonMatchingValue`, you can call it something more meaningful for the test - e.g. `ValueToQueryFor`.
2. Instead of having 3 different tests for matches, non-matches due to value difference and non-matches due to type difference, you can add `ExpectedMatches` field to the test object.

Reply:
So you'd like to have only one unit test, for the sake of argument called `QueryFilter_WithNumericArguments`, which does not specify in the name what is the expected outcome (shouldThrow, shouldMatch, shouldNotMatch), instead of 

1. `QueryFilter_WithImplicitlyConvertibleNumericArguments_ShouldMatch`
2. `QueryFilter_WithImplicitlyConvertibleNumericArguments_ShouldNotMatch`
3. `QueryFilter_WithNonImplicitlyConvertibleArguments_ShouldNotMatch`

Did I understand correctly?

Reply:
yep